File: tecplot10/util/tecio/simtest.f
C
C Simple example fortran program to write a
C binary datafile for tecplot.  This example
C does the following:
C
C   1.  Open a datafile called "t.plt"
C   2.  Assign values for X,Y, and P
C   3.  Write out a zone dimensioned 4x5
C   4.  Close the datafile.
C
C
      program test

      INCLUDE 'tecio.inc'

      character*1 NULLCHR
      Integer*4   Debug,III,NPts,NElm

      Dimension X(4,5), Y(4,5), P(4,5)
      Integer*4 TecIni,TecDat,TecZne,TecNod,TecFil
      Integer*4 VIsDouble

      NULLCHR = CHAR(0)
      Debug   = 1
      VIsDouble = 0
      IMax    = 4
      JMax    = 5
      KMax    = 1
C
C... Open the file and write the tecplot datafile 
C... header information.
C
      I = TecIni('SIMPLE DATASET'//NULLCHR,
     &           'X Y P'//NULLCHR,
     &           't.plt'//NULLCHR,
     &           '.'//NULLCHR,
     &           Debug,
     &           VIsDouble)

      Do 10 I = 1,4
      Do 10 J = 1,5
        X(I,J) = I
        Y(I,J) = J
        P(I,J) = I*J
   10 Continue
C
C... Write the zone header information.
C
      I = TecZne('Simple Zone'//NULLCHR,
     &           IMax,
     &           JMax,
     &           KMax,
     &           'BLOCK'//NULLCHR,
     &           NULLCHR)
C
C... Write out the field data.
C
      III = IMax*JMax
      I   = TecDat(III,X,0)
      I   = TecDat(III,Y,0)
      I   = TecDat(III,P,0)

      I = TecEnd()
      Stop
      End

File: tecplot10/util/tecio/simtest.c
/*
 * Simple example c program to write a
 * binary datafile for tecplot.  This example
 * does the following:
 *
 *   1.  Open a datafile called "t.plt"
 *   2.  Assign values for X,Y, and P
 *   3.  Write out a zone dimensioned 4x5
 *   4.  Close the datafile.
 */

#include "TECIO.h"

#ifndef NULL
#define NULL 0
#endif

main ()
{
  float X[5][4], Y[5][4], P[5][4];
  INTEGER4 Debug,I,J,III,/*NPts,NElm,*/DIsDouble,VIsDouble,IMax,JMax,KMax;

  Debug     = 1;
  VIsDouble = 0;
  DIsDouble = 0;
  IMax      = 4;
  JMax      = 5;
  KMax      = 1;
/*
 * Open the file and write the tecplot datafile 
 * header information 
 */
  I = TECINI("SIMPLE DATASET",
             "X Y P",
             "t.plt",
             ".",
             &Debug,
             &VIsDouble);

  for (J = 0; J < 5; J++)
  for (I = 0; I < 4; I++)
    {
      X[J][I] = (float)(I+1);
      Y[J][I] = (float)(J+1);
      P[J][I] = (float)((I+1)*(J+1));
    }
/*
 * Write the zone header information.
 */
  I = TECZNE("Simple Zone",
             &IMax,
             &JMax,
             &KMax,
             "BLOCK",
             NULL);
/*
 * Write out the field data.
 */
  III = IMax*JMax;
  I   = TECDAT(&III,&X[0][0],&DIsDouble);
  I   = TECDAT(&III,&Y[0][0],&DIsDouble);
  I   = TECDAT(&III,&P[0][0],&DIsDouble);

  I = TECEND();
}
